Stable release HYPE

Dougley opened the pull request.

Arzte committed:
Does it work?

Arzte committed:
Move all discord specific configs to one discord config vaule.

To keep with the theme of how uservoice is layed out.

Arzte committed:
Beutify all the code.

Arzte committed:
Add sub domain and domain configuration keys.

This is for UserVoice.

Arzte committed:
EsLint Fix-all changes

Arzte committed:
Move bot const to bottom of const.

This looks better than it being in the middle.

Arzte committed:
Forgot to change the discord token

I forgot to change it after I changed the config variables some.

Arzte committed:
UserVoice working in client

It eventually took me relizing that the uservoice API thingy needed me to use .get(<get_stuff_goes_here>) on uvclient to get data pulled.

Arzte committed:
Add functions

Its not pretty outputs yet, but we'll get there.

Arzte committed:
Add a startup message

This is so we know when the bot is done starting and has connected to the API.

Arzte committed:
Start making working functions for UserVoice

Arzte committed:
Get a nice embed with the list of suggestions.

Arzte committed:
Update the bot code for a new server name.
TODO: Change this to be more flexable

Arzte committed:
Add rudimentary search functionality. Also add sendTyping for the delay it takes to return info.
Search still needs changes that Dabbit suggested in the Private testing server for the bot.

Arzte committed:
Update .gitignore with test files
The files that this ignores are .json files containting a json api response to make it easier for me to make embeds with the response.

Arzte committed:
Make UV stuck at 0.0.11 until manually updated.

Arzte committed:
Add npm start script

Arzte committed:
Add a message cache to keep the memory low.
It hasn't been an issue, just preparing for the future.

Arzte committed:
Fully implment search functionality.

Arzte committed:
Fix Grammer Mistakes/Less than 5 doesn't show full list of suggestions

Arzte committed:
Comment Code
This is to make contributing easier, and to make it easier to remeber how something worked later on.

Arzte committed:
Update README.md

Arzte committed:
Add logging
idk why we need it, Dabbit wanted it for some reason.

Arzte committed:
Update UserVoice version

Arzte committed:
Add comment command, Change bot logging
Comment command allows for discord users to comment on UserVoice suggestions via. Discord.
Bot logging was changed to only show details important for moderating, rather than just everything.
Also do some eslint fixes.

Arzte committed:
Custodians & above are the only people allowed to use email for the comment.
This makes it so the average user isn't required to enter in a email. (We plan to get the users email from UserVoice, will be doing that later)

Arzte committed:
Improve error handling, move email to config
Default email*

Arzte committed:
Grab the users email from UV based on snowflake.
This is so we can use UserVoice's loginAs API feature, which allows us to loginAs any UserVoice account on the instance. Since Discord has the GUID set the same as Discord, we can easily pull the email from UserVoice.

Arzte committed:
Fix a few errors with email detection

Arzte committed:
Change to 4 spaces for indent for better readablity
Also specify the enviroment is node.
Add test.js to .gitignore so it wouldn't get picked up by git or eslint. (I use that file to temp. hold js pastings)

Arzte committed:
Add the vote command.
Make a few things in the embed clearer, add a url for people who disabled website previews.
Handle 404 errors from UserVoice.
Switch to console.error() whenever we'd be logging something to console as a result of an error.

Arzte committed:
catch loginAs errors (if there ever are any)

Arzte committed:
Make vote work and easier to unstand how to use.

Arzte committed:
Removed test.js from being tracked
Its just a file for me to copy code to if I need to copy something else before pasting the code back.

Arzte committed:
Tell eslint to buzz off
We aren't using the V2 of UserVoice ATM but we want to leave it in for when we do.  This just makes eslint ignore the fact that v2Client isn't used anywhere.

Dougley committed:
Submit command

And some cleanup

Dougley committed:
Small bit of logging

Arzte committed:
Remove useless code, add error handling,
also send url with embed.

Dougley committed:
Initial stump for duplicate command

Dougley committed:
Approval queue stuff

Dougley committed:
Deny/approve commands

Dougley committed:
Approval queue stuff

Still need to do actual things with report, right now they just void out
when approved/denied.

Dougley committed:
Small tweaks

Dougley committed:
Delete cards from UV when duplicate report passed

Getting close to a beta test I'd recon!

Dougley committed:
Forgot somethings

Dougley committed:
Duplicate reports are only to be approved by admins

Dougley committed:
Send new cards to the queue for review

PUBLIC BETA HYPE

Dougley committed:
Left a debug statement in

Dougley committed:
Comment out debug thingsy

Commit:
Add message separator

Dougley committed:
Merge pull request #2 from nullpixel1/patch-1

Add message separator

Dougley committed:
Fix some of the issues that were reported

Dougley committed:
No useless subcommands pls

Commit:
Added separator to queue (#3)

* Added separator to queue

* Update some grammar and remove log seperators

Dougley committed:
Change varibales to support the removed subcommand

nullpixel committed:
Timeout removing messages (#4)

* No emebd pls

* Removed that annoying space

* Dabbit request: timeout removing messages

* :face_palm: I'm pretty stupid

* Fix the undefined delay issue

* Only return a promise

* Added config option for delay time

* Missed a few

* Fix undefined arg

I'm sorry for the hastle and I will get my own instance up so I can test it next time

* Rename deletThis() to deleteThis()

@dougly needs to write better function names *cough* *cough*

Dougley committed:
Tweaked some things to support changes requested

Dougley committed:
Admin queue support for duplicate reports

Dougley committed:
Channel binds

Dougley committed:
Fix poor formatting

Dougley committed:
Error out when the message will be too long

nullpixel committed:
Final changes (#5)

* Alert when user isn't signed in

* Detailed error messages

* Search is done, and beautified, pls no hate

Dougley committed:
Offline user hack

nullpixel committed:
Fixed multiple trello issues (#7)

* Alert when user isn't signed in

* Detailed error messages

* Search is done, and beautified, pls no hate

* w00t, fixed https://trello.com/c/DXQ5ZxDN/38-users-can-approve-and-deny-the-same-report, https://trello.com/c/XrfwBJtJ/37-users-can-approve-their-own-feedback

* fixed The error message for an incorrect report ID is wrong on

* You can mark the same suggestion as duplicates of each other.

nullpixel committed:
Fix last issue (#8)

* Alert when user isn't signed in

* Detailed error messages

* Search is done, and beautified, pls no hate

* w00t, fixed https://trello.com/c/DXQ5ZxDN/38-users-can-approve-and-deny-the-same-report, https://trello.com/c/XrfwBJtJ/37-users-can-approve-their-own-feedback

* fixed The error message for an incorrect report ID is wrong on

* You can mark the same suggestion as duplicates of each other.

* FIX THE LAST ISSUE FOR RELEASE W00T

* fix stupid idotic copypasaatrta

Dougley committed:
Oversight

SirDoctors commented:
Lots to trawl through, will be working on reviewing this in chunks. This will take some time and due to the way GitHub does reviews my comments for said review won't show as they are made.

Dougley committed:
Fix bugs

Dougley committed:
Data retention

Arzte committed:
Allow URL's in comments.

Allows for stuff like https://feedback.discordapp.com/forums/326712-discord-dream-land/suggestions/18661513-translator-badges as suggestion to comment as.

Dougley committed:
Restart command

nullpixel committed:
Fix minor bugs (#10)

* Alert when user isn't signed in

* Detailed error messages

* Search is done, and beautified, pls no hate

* w00t, fixed https://trello.com/c/DXQ5ZxDN/38-users-can-approve-and-deny-the-same-report, https://trello.com/c/XrfwBJtJ/37-users-can-approve-their-own-feedback

* fixed The error message for an incorrect report ID is wrong on

* You can mark the same suggestion as duplicates of each other.

* FIX THE LAST ISSUE FOR RELEASE W00T

* fix stupid idotic copypasaatrta

* fixing minor bugs

* fix dougley request:tm:

Dougley committed:
smh null

Dougley committed:
Merge remote-tracking branch 'origin/uservoice' into uservoice

Dougley committed:
smh

Dougley committed:
Restore old version of comment

Newer version was broken and I was too lazy to properly debug it.

Dougley committed:
Fix issues

And add bugsnag

SteamingMutt commented:
Considering the project is being overhauled, and the current version isn't fit for production, I'm closing this PR.
